### ISSUE TITLE
MultiLogloss metric strange results

### ISSUE THREAD

**Issue by FiksII** [catboost/catboost#2236]
Catboost version: 1.1.1
Operating System: Ubuntu 20.04

Problem: I train a model for multilabel classification
```
from sklearn import datasets
import catboost
from catboost.utils import eval_metric
from sklearn.model_selection import train_test_split

X, y = datasets.make_multilabel_classification(n_samples=500, n_features=20, n_classes=5, random_state=0)
X_train, X_test, y_train, y_test = train_test_split(X, y)

data = catboost.Pool(X_train, y_train)
eval_data = catboost.Pool(X_test, y_test)

classifier = catboost.CatBoostClassifier(iterations=5, learning_rate=0.05, min_data_in_leaf=1024, used_ram_limit='60Gb',
                                         max_ctr_complexity=1, border_count=None, loss_function='MultiLogloss',
                                         eval_metric='MultiLogloss')

classifier.fit(data, eval_set=eval_data, logging_level='Verbose', early_stopping_rounds=10)

print(classifier.best_score_)
```

>0:	learn: 0.6792034	test: 0.6810916	best: 0.6810916 (0)	total: 1.8ms	remaining: 7.22ms
1:	learn: 0.6625325	test: 0.6709725	best: 0.6709725 (1)	total: 3.52ms	remaining: 5.29ms
2:	learn: 0.6485148	test: 0.6624420	best: 0.6624420 (2)	total: 5.28ms	remaining: 3.52ms
3:	learn: 0.6332720	test: 0.6553559	best: 0.6553559 (3)	total: 6.97ms	remaining: 1.74ms
4:	learn: 0.6220026	test: 0.6486661	best: 0.6486661 (4)	total: 8.58ms	remaining: 0us

> 
> bestTest = 0.6486660866
> bestIteration = 4
> 
> {'learn': {'MultiLogloss': 0.622002633033872}, 'validation': {'MultiLogloss': 0.6486660866018348}} 

But, when I want to calculate test metrics directly, I see the strange results
```
y_predict = classifier.predict_proba(X_test)

print(eval_metric(label=y_test, approx=y_predict, metric='MultiLogloss'))
print(catboost.metrics.MultiLogloss().eval(label=y_test, approx=y_predict))
```
>[0.7733648278031129]
[0.7733648278031129]

Why do we have here 0.77, but during the train, it was 0.64? What am I doing wrong?
P.S. HammingLoss is ok.
